apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/codegen/java/helpers/NamedType.kt

```diff
@@ -26,11 +26,13 @@ internal class NamedType(
 
 
 internal fun NamedType.toParameterSpec(context: JavaContext): ParameterSpec {
+  val irType = context.resolver.resolveIrType(type)
   return ParameterSpec
       .builder(
-          context.resolver.resolveIrType(type),
+          irType.withoutAnnotations(),
           context.layout.propertyName(graphQlName),
       )
+      .addAnnotations(irType.annotations)
```

Comment:
Is that a Java8 thing? (https://docs.oracle.com/javase/8/docs/api/java/lang/annotation/ElementType.html#TYPE_USE)

Asked differently: does the generated code require Java8 compatibility? (I think it is fine if that's the case, just trying to wrap my head around the implications there and planning for future Java19 work 🙃 )

Reply:
Nice. I'd be curious to see what happens trying to compile the generated sources with Java7 ...

Reply:
[Tried it](https://github.com/apollographql/apollo-kotlin/tree/compile-against-java-1-7), didn't work:

```
    Compilation produced the following diagnostics:
    build/generated/test/hero_with_review/java/operationBased/com/example/hero_with_review/selections/TestQuerySelections.java:21: error: incompatible types: java.util.List<com.apollographql.apollo3.api.CompiledField> cannot be converted to java.util.List<com.apollographql.apollo3.api.CompiledSelection>
      private static List<CompiledSelection> __createReview = Arrays.asList(
```

Reply:
I guess that means `TargetLanguage.JAVA` should be `TargetLanguage.JAVA8`